File: Disciplinas/AED/AED2/labs/lab01/sem-funcao/q02.java
import java.util.Scanner;

class q02 {

    public static boolean verificaFim(String str) {
		boolean ehFim = false;

		if(str.charAt(0) == 'F' && str.charAt(1) == 'I' && str.charAt(2) == 'M') ehFim = true; 
		
		return ehFim;
	}

	public static int contarMaiusculas(String str, int i) {
		int qt = 0;

        if(i >= str.length()) qt = 0;
        else{
		    if(str.charAt(i) >= 65 && str.charAt(i) <= 90) qt = 1 + contarMaiusculas(str, i+1);
            else qt = 0 + contarMaiusculas(str, i+1);
        }

		return qt;
	}

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        String str;

        do{

            str = scan.nextLine();
			if(!verificaFim(str)) {
				System.out.println(contarMaiusculas(str, 0));
			}

        }while(!verificaFim(str));

        scan.close();
    }
}

File: AED/AED2/labs/lab01/sem-funcao/q02.c
#include <stdio.h>

int verificaFim(char* str) {
	int ehFim = 0;
	if(str[0] == 'F' && str[1] == 'I' && str[2] == 'M' && str[3] == '\0') ehFim = 1;

	return ehFim;
}

void removerQuebraLinha(char* str) {

	for(int i = 0; str[i] != '\0'; i++) {
		if(str[i] == '\n') {
			str[i] = '\0';
			break;
		}
	}
}

int contarMaiusculas(char* str, int i) {
    int qt = 0;

    if(str[i] == '\0') qt = 0;
    else {
        if(str[i] >= 65 && str[i] <= 90) qt = 1 + contarMaiusculas(str, i+1);
        else qt = 0 + contarMaiusculas(str, i+1);
    }

    return qt;
}

int main(void) {

    char str[100];

    do {

        fgets(str, 100, stdin); 
        removerQuebraLinha(str);

        if(!verificaFim(str) && str[0] != '\0') {
			printf("%d\n", contarMaiusculas(str, 0));	
		}


    } while(!verificaFim(str));

    return 0;
}